canine/adapters/base.py

```diff
@@ -3,6 +3,43 @@
 from itertools import product, repeat
 from functools import reduce
 
+class _FixedArray(object):
+    """
+    Helper to capture arrays which are marked as fixed
```

Comment:
Make it more clear that FixedArrays must be 2D. BTW, should we catch this in the constructor, rather than in `__getitem__` or `__iter__`?